### ISSUE TITLE
Client-side and server-side errors are difficult to differentiate

### ISSUE THREAD

**Issue by thully** [socketio/socket.io-client-swift#679]
In socket.io-client-swift, both client-side errors (i.e. issues connecting to the server or maintaining a connection) and server-side errors (i.e. an Error actually emitted by the server) are handled by the same "error" handler. This sometimes makes it hard to differentiate, and is different from how other socket.io clients function (on socket.io-client-java, there are separate "connect error" and "reconnect error" events). Could these events be separated in a future version?

**Comment by nuclearace** [comment]
Yeah I was thinking the same thing. It's weirder with the changes in https://github.com/socketio/socket.io-client-swift/commit/dd2167a09f8b2e1dc125ad9f6a07bbe7c578c588.

The only issue is that would be major version bump.

**Comment by wongzigii** [comment]
@nuclearace Hello, is there any plan for this issue in the next release?

**Comment by 2fours** [comment]
This is how I differentiate disconnects for now (v13.2.0) until there's concrete status codes or something:

```
[self.socket.defaultSocket on:@"disconnect" callback:^(NSArray * data, SocketAckEmitter * ack) {
    //server disconnect seems to yield data of "Socket Disconnected" vs client with "Disconnect" so only attempt reconnect when server disconnects
    id object0 = [data objectAtIndex:0];
    if ([object0 isEqualToString:@"Socket Disconnected"]) {
        [self reconnect];
    }
}];
```

**Comment by wongzigii** [comment]
@2fours This will break if the string change in the future release.

**Comment by 2fours** [comment]
@wongzigii Luckily we're in control of when to update the libraries we depend on and can check to see if things are broken. Perhaps one could even write a unit test. Also me saying "until there's concrete status codes or something" hints that I'm aware of obvious issues like the one you point out.
